src/nvim/ui_bridge.c

```diff
@@ -181,12 +182,19 @@ static void ui_bridge_cursor_goto_event(void **argv)
 
 static void ui_bridge_cursor_styleset(UI *b, Dictionary args)
 {
-  UI_CALL(b, cursor_styleset, 2, b, &args);
+  Object copy = copy_object(DICTIONARY_OBJ(args));
+  Object *a = xmalloc(sizeof(Object));
+  *a = copy;
+  UI_CALL(b, cursor_styleset, 2, b, a);
```

Comment:
The event can be deferred apparently in tui_scheduler so you have to make a copy else the original args is already freed when the event triggers.
*a = copy should work just fine in C, struct assignment is equivalent to a memcpy (?!): the deepcopy is done via the copy_object before.